File: src/main/java/br/gov/prefeitura/doacoes/entity/Institution.java
package br.gov.prefeitura.doacoes.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "institutions")
public class Institution extends BaseEntity {

    @Column(nullable = false, length = 150)
    private String name;

    @Column(nullable = false, unique = true, length = 20)
    private String cnpj;

    @Column(nullable = false, length = 200)
    private String address;

    @Column(length = 20)
    private String phone;

    @Column(length = 150)
    private String email;

    @Column(name = "responsible_name", length = 150)
    private String responsibleName;

    @Column(name = "families_served")
    private Integer familiesServed;

    @Column(nullable = false)
    @Builder.Default
    private Boolean active = true;

}

File: frontend/src/types/institution.ts
export interface Institution {
  id: number;
  name: string;
  cnpj: string;
  address: string;
  phone: string | null;
  email: string | null;
  responsibleName: string | null;
  familiesServed: number | null;
  active: boolean;
}

export interface InstitutionRequest {
  name: string;
  cnpj: string;
  address: string;
  phone?: string;
  email?: string;
  responsibleName?: string;
  familiesServed?: number;
  active?: boolean;
}
